packages/tfchain_client/src/farms.ts

```diff
@@ -89,7 +90,16 @@ class Farms extends QueryFarms {
   @checkConnection
   async removeFarmIp(options: RemoveFarmIPOptions) {
     const extrinsic = this.client.api.tx.tfgridModule.removeFarmIp(options.farmId, options.ip);
-    return this.client.patchExtrinsic(extrinsic);
+    return this.client.patchExtrinsic<void>(extrinsic);
+  }
+
+  @checkConnection
+  async removeFarmIps(options: RemoveFarmIPOptions[]) {
+    const extrinsics: ExtrinsicResult<void>[] = [];
+    for (const option of options) {
+      extrinsics.push(await this.removeFarmIp(option));
+    }
+    await this.client.applyAllExtrinsics<void>(extrinsics);

```

Comment:
this method shouldn't be added to the chain client, the optimal place is [here](https://github.com/threefoldtech/tfgrid-sdk-ts/tree/development/packages/grid_client/src/clients/tf-grid). please add a new file for the farms and implement it there.

this is an [example](https://github.com/threefoldtech/tfgrid-sdk-ts/blob/development/packages/grid_client/src/clients/tf-grid/contracts.ts#L319-L326) for deleting the contracts with a batch call.

Reply:
function created